### ISSUE TITLE
does the range of cards shrink with each round ?

### ISSUE THREAD

**Issue by nguyenviettuan96** [bupticybee/TexasSolver#159]
I mean if I run your algo from river and I run your algo from flop with the same community card, oop range and ip range. And the starting pot at river same for 2 runs. Is that the result same ?
If we solve from river not from flop --> river, the result maybe wrong

**Comment by bupticybee** [comment]
> I mean if I run your algo from river and I run your algo from flop with the same community card, oop range and ip range. And the starting pot at river same for 2 runs. Is that the result same ? If we solve from river not from flop --> river, the result maybe wrong

You are using it in the wrong way. You can't just keep the range unchanged and except the result to be the same. 

Game theory just don't work that way.

You sulely will get different results. It's a game theory feature.

**Comment by bupticybee** [comment]
I'm not sure how to explan it to you without involving algorithms and game theory stuff.

**Comment by nguyenviettuan96** [comment]
yeah I understanded that is what I confuse when I implement your code in river round in real time

**Comment by bupticybee** [comment]
> yeah I understanded that is what I confuse when I implement your code in river round in real time. I calculate that if we solve (turn + river) for 1755 different flops, we need at least 68455 diffrent situations, so with 89 seconds for each, usual core i5 could need 70 days. That is impossible, so real time is what I must consider

I think what I can suggest is to use fewer bet sizes and raise sizes, and decrease range density or find a way to decrease pot depth. These are key elements to reduce computation time in any solver.

**Comment by nguyenviettuan96** [comment]
btw could u tell me what is isomorphism
